src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php

```diff
@@ -61,9 +61,12 @@ public function get(RequestConfiguration $requestConfiguration, RepositoryInterf
 
         if ($paginator instanceof Pagerfanta) {
             $request = $requestConfiguration->getRequest();
+            if (null !== $limit = $request->query->get('limit')) {
+                $limit = (int)$limit;
+            }
 
             $paginator->setMaxPerPage($this->resolveMaxPerPage(
-                $request->query->get('limit'),
+                $limit,
```

Comment:
Won't the `$request->query->getInt('limit', null)` work?

Reply:
hah, i didn't see that method. will update

Reply:
@lchrusciel done

Reply:
Didn't know that it will break so many things... sorry :(

Reply:
I'll take a look